### ISSUE TITLE
Output of message differs on dev machine build server

### ISSUE THREAD

**Issue by DavidSSL** [shouldly/shouldly#234]
On my dev machine, I get the following error message:

  Shouldly.ShouldAssertException : 
        factorySettingOverrideResult should satisfy all the conditions specified, but does not.
        The following errors were found ...
--------------- Error 1 ---------------

    factorySettingOverrideResult
						.DemoModes.Distinct()
						.Count()
        should be
    2
        but was
    3

However, on the build server, I get: 

Test(s) failed. Shouldly.ShouldAssertException : 
        The provided expression should satisfy all the conditions specified, but does not.
        The following errors were found ...
--------------- Error 1 ---------------

    The provided expression
        should satisfy all conditions
    2
        but was
    3

I'm using Shouldly 2.5.0. Ideally I'd like the same output that is in my dev output to be shown on the build server. So why are the output different?

**Comment by JakeGinnivan** [comment]
Shouldly uses the source code to make it's error messages better. So this issue is likely to be one of two issues:

1) You are building tests in Release module without PDB's
2) You are building your test assembly in another build, and running tests in a different one so the source which the test assemblies were built from are not available.

Without the source we fall back to the best we can do (which is about the same as most other frameworks)

**Comment by DavidSSL** [comment]
Ok, we've found the issue. Effectively in our release build we have got this:

![image](https://cloud.githubusercontent.com/assets/483473/7455651/6a14fd7c-f27e-11e4-9aa7-1fb120e73c43.png)

but in the debug/development build we've got this:

![image](https://cloud.githubusercontent.com/assets/483473/7455673/86da6690-f27e-11e4-8e84-8eb5ad8fbdd4.png)

and this difference in "Debug Info" caused the issue. Hence, I think that it's not enough to simply say that the pdb's should be present. It might also be a good idea to the update the relevant documentation (Point out where that would be and I can update it).
